### PR TITLE
Adds 'typesVersions' support to NodeJS definitions

### PR DESCRIPTION
**Pull request by rbuckton**
This adds `"typesVersions"` support to the NodeJS definitions and supersedes #30477

### CODE REVIEW

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -142,34 +142,8 @@ interface ErrorConstructor {
...
-// Forward-declare needed types from lib.es2015.d.ts (in case users are using `--lib es5`)
-interface Iterable<T> { }
-interface Iterator<T> {
-    next(value?: any): IteratorResult<T>;
-}
-interface IteratorResult<T> { }
-interface IterableIterator<T> { }
-interface AsyncIterableIterator<T> {}
 interface SymbolConstructor {
     readonly observable: symbol;
```
Comment (rbuckton) [inline types/node/globals.d.ts]: I'd have to look to see when this was added, but this value does not actually exist in NodeJS and most likely should also be removed.

**Comment by SimonSchick** [comment]
This brings up the question, when we make PRs to node typings now, do we need to update *both* 2.x and 3.x typings?
I personally thing we should deprecate the 2.x typings and tell people to move forward with 3.x

**Comment by rbuckton** [comment]
@DanielRosenwasser what is your take on @SimonSchick's recommendation?

BTW: The build is failing only due to the fact dtslint does not currently handle the changes for `readonly` arrays introduced in the latest development builds and is otherwise good.

**Comment by DanielRosenwasser** [comment]
I think it's fine to encourage people to update both when possible, but not have a specific mandate that they update everything. @RyanCavanaugh may have a better sense for deprecation policy than I do.

**Comment by Flarna** [comment]
Seems @typescript-bot missed this issue, similar as it missed #30477. Also the label added "New Definition" is wrong. Could someone tell @typescript-bot that typesVersions is supported now?

Regarding change itself: will take a look later...

**Comment by rbuckton** [comment]
@Flarna, I filed https://github.com/RyanCavanaugh/dt-mergebot/issues/13 for that issue.

File: types/node/globals.d.ts (hunk 1 above)
Comment (Flarna) [reply types/node/globals.d.ts]: see #15093, #26034 (observable) and #25269 (asyncIterator)
Not sure if the rxjs (v6) compatibility is still needed.

**Review by SimonSchick — APPROVED** [review]

**Comment by jamiekyle-eb** [comment]
I'm getting the following error after this commit in `typescript@3.3.3`:

```js
node_modules/@types/node/repl.d.ts:96:43 - error TS2709: Cannot use namespace 'InspectOptions' as a type.

96     const writer: REPLWriter & { options: InspectOptions };
                                             ~~~~~~~~~~~~~~
```

> Note: This error message is confusing, it's just saying that `InspectOptions` does not exist. [See here](https://github.com/Microsoft/TypeScript/issues/27311#issuecomment-424023704)

It seems to be picking up `node/ts3.1/util.d.ts` but not doing declaration merging with `node/util.d.ts`.

**Comment by Flarna** [comment]
@co3moz see #33098

**Comment by Flarna** [comment]
@jamiekyle-eb Do you have some more context, a small reproducer? Maybe it's better to open a dedicated issue instead adding more comments on a merged PR.

**Pull request merged**